graphql.go

```diff
@@ -208,16 +208,16 @@ func (c *Client) runWithPostFields(ctx context.Context, req *Request, resp inter
 	if err != nil {
 		return err
 	}
+	if res.StatusCode != http.StatusOK {
```

Comment:
What do you think about creating a custom error type to encapsulate status code, response body and response headers? I think that will give the caller more info to take appropriate actions.

Reply:
Absolutely! Given it extends the original error interface I'm guessing it should not be a breaking change for our existing clients